waku/node/v2/wakunode2.nim

```diff
@@ -97,17 +97,7 @@ proc init*(T: type WakuNode, nodeKey: crypto.PrivateKey,
     filters: initTable[string, Filter]()
   )
 
-  # TODO This is _not_ safe. Subscribe should happen in start and after things settled down.
-  # Otherwise GRAFT message isn't sent to a node.
-  for topic in topics:
-    proc handler(topic: string, data: seq[byte]) {.async, gcsafe.} =
-      debug "Hit handler", topic=topic, data=data
-
-    # XXX: Is using discard here fine? Not sure if we want init to be async?
-    # Can also move this to the start proc, possibly wiser?
-    discard result.subscribe(topic, handler)
-
-proc start*(node: WakuNode) {.async.} =
+proc start*(node: WakuNode, topics = newSeq[string]()) {.async.} =
```

Comment:
Hm so we are changing the signature of `start`. It has a default argument which is good. API docs still have to be updated.

Is this the best way to do this in general? It might be, just not sure. What do you think @kdeme? Perhaps we could do it like all other "started node actions" and do it right after start in main?

Reply:
@oskarth so the reason why I did it here is that we get this functionality elsewhere too, eg chat. Otherwise we need to reimplement the wait there too if it is something thats part of the main module.

Reply:
I think it makes sense to do this here.
But it depends on how much flexibility we want/need. Do we really need to be able to subscribe to different PubSub topics at the same time? Meaning, having different "messaging networks" where we publish and subscribe to?

(I'm not really talking about partitioning the topics, this could be arranged "underneath" I believe)

Reply:
>  Do we really need to be able to subscribe to different PubSub topics at the same time with different handlers?

Yes, I believe so. Basic example: node that supports Status app using a Status topic, and another WalletConnect that uses another topic (completely separate from Status Network).

Reply:
That said, I can see reasoning above and think we can probably do it here. In the case of multiple networks this can be probably be done with API and/or handler-of-handler and/or extended later as we see specific needs.

API docs still needs updating.